pkg/l2announcer/l2announcer.go

```diff
@@ -311,8 +311,19 @@ func (l2a *L2Announcer) processPolicyEvent(ctx context.Context, event resource.E
 }
 
 func (l2a *L2Announcer) upsertSvc(svc *slim_corev1.Service) error {
-	// Ignore services managed by an unsupported load balancer class.
 	key := serviceKey(svc)
+
+	// Ignore services if there is no noExternal or LB IP assigned.
+	noExternal := svc.Spec.ExternalIPs == nil
+	noLB := true
+	for _, v := range svc.Status.LoadBalancer.Ingress {
+		noLB = noLB && v.IP == ""
```

Comment:
This took me a bit to parse - I haven't seen this sort of assignment used before. Is this a common Cilium pattern?

Reply:
Thanks for pointing it out. Changed it.